packages/cli/src/scripts/codemod/transforms/v2/main.cts

```diff
@@ -347,6 +347,75 @@ export default function transform(file: FileInfo, api: API, options?: Options):
           headerContentPinnable.remove();
           isDirty = true;
         }
+
+        const headerContent = j(el).find(j.JSXAttribute, { name: { name: 'headerContent' } });
+        if (headerContent.size() > 0) {
+          const attr = headerContent.get();
+          if (attr.value.value && attr.value.value.type === 'JSXExpressionContainer') {
+            headerContent.forEach((el) => {
+              const dynamicPageHeader = j(el).find(j.JSXElement, {
+                openingElement: { name: { name: 'DynamicPageHeader' } }
+              });
+
+              // remove DynamicPageHeader import only if no DynamicPage is there
+              if (!componentIsImportedFromWebComponentsReact(j, root, 'DynamicPage')) {
+                const imports = root.find(j.ImportDeclaration);
+                const dptImport = imports.find(j.ImportSpecifier, { local: { name: 'DynamicPageHeader' } });
+                dptImport.remove();
+              }
+              if (!componentIsImportedFromWebComponentsReact(j, root, 'ObjectPageHeader')) {
```

Comment:
This check is not needed, the addWebComponentsReactImport checks already internally